maml_rl/samplers/multi_task_sampler.py

```diff
@@ -294,39 +323,67 @@ def create_episodes(self,
         episodes.log('process_name', self.name)
 
         t0 = time.time()
+        #print("create episode")
+        if params is not None:
+            old_params = self.agent.policy_net.state_dict()
+            self.agent.policy_net.load_state_dict(params, strict=False)
         for item in self.sample_trajectories(params=params):
             episodes.append(*item)
         episodes.log('duration', time.time() - t0)
-
-        self.baseline.fit(episodes)
-        episodes.compute_advantages(self.baseline,
+        self.baseline.fit(episodes, self.agent)
+        episodes.compute_advantages(self.baseline, self.agent,
                                     gae_lambda=gae_lambda,
                                     normalize=True)
+        if params is not None:
+            self.agent.policy_net.load_state_dict(old_params) # hope there is no race condition here! NOTE: strict=True
```

Comment:
@MarcCote i think this is fundamentally different than tristan's corresponding code :
1) He is using `params` on the fly for the policy net.
2) We need `params` for baseline. 

Comments requested on this.

Reply:
Right, so having the strings already embedded, you don't need the policy's params.